expertsystem/data.py

```diff
@@ -183,14 +177,14 @@ def __init__(  # pylint: disable=too-many-arguments
         self,
         name: str,
         pid: int,
-        state: ParticleQuantumState,
+        state: QuantumState[float],
         mass: float,
         width: float = 0.0,
     ):
         super().__init__(complex(mass, width))
         self.__name: str = name
         self.__pid: int = pid
-        self.state: ParticleQuantumState = state
+        self.state: QuantumState[float] = state
```

Comment:
We may want to do a check on the argument here, like `not isinstance(state, Spin)`. Because the user does have to be familiar with the syntax when creating a new particle.

Might be addressed by #152, otherwise, see remark on 'default' template argument.

Reply:
Hmm but mypy is already catching those errors. So how can the instance check help?

Reply:
`mypy` does not run on notebooks, which is the most likely place where someone wants to construct a new particle outside the framework.

Reply:
Ok, I totally missed that. But thats a huge problem. Basically the complete UI has to be protected with isinstance calls as well then

Reply:
Yeah, this touches on a classic Python discussion, I guess, that one assumes a certain responsibility with the user. So we could assume the user not to misuse things here, it's just because of this uncommon template syntax that the change for mistakes is larger.
For sure, #152 would simplify things on a notebook side, so could leave it.